sway-lib-std/src/identity.sw

```diff
@@ -3,8 +3,18 @@ library identity;
 
 use ::address::Address;
 use ::contract_id::ContractId;
+use ::intrinsics::size_of_val;
+use ::mem::{addr_of, eq};
 
 pub enum Identity {
     Address: Address,
     ContractId: ContractId,
 }
+
+impl core::ops::Eq for Identity {
+    fn eq(self, other: Self) -> bool {
+        match(self, other) {
+            (Identity::Address(address1), Identity::Address(address2)) => eq(addr_of(address1), addr_of(address2), size_of_val(self)), (Identity::ContractId(asset1), Identity::ContractId(asset2)) => eq(addr_of(asset1), addr_of(asset2), size_of_val(self)), _ => false, 
```

Comment:
Instead of using `mem::eq` here, why not just use `address1 == address2`? Address implements `Eq`: https://github.com/FuelLabs/sway/blob/7fb883c029fa008887b413dc95c2763a4cc73887/sway-lib-std/src/address.sw#L13

You might have to import `::address::*`.